File: python-client/goredis/transactions.py
from .client import get_client

def Multi():
    """Mark the start of a transaction block."""
    return get_client().send_command("MULTI")

def Exec():
    """Execute all commands queued in a transaction."""
    return get_client().send_command("EXEC")

def Discard():
    """Discard all commands issued after MULTI."""
    return get_client().send_command("DISCARD")

def Watch(*keys):
    """Watch the given keys to determine execution of the MULTI/EXEC block."""
    return get_client().send_command("WATCH", *keys)

def Unwatch():
    """Forget about all watched keys."""
    return get_client().send_command("UNWATCH")

File: go-client/transactions.go
package goredis

// Multi starts a transaction block.
func Multi() (interface{}, error) {
	return mustGetClient().SendCommand("MULTI")
}

// Exec executes all the commands in the transaction block.
func Exec() (interface{}, error) {
	return mustGetClient().SendCommand("EXEC")
}

// Discard flushes all previously queued commands in a transaction block.
func Discard() (interface{}, error) {
	return mustGetClient().SendCommand("DISCARD")
}

// Watch watches the given keys for modifications.
func Watch(keys ...string) (interface{}, error) {
	cmdArgs := append([]interface{}{"WATCH"}, toInterfaceSlice(keys)...)
	return mustGetClient().SendCommand(cmdArgs...)
}

// Unwatch flushes all the watched keys.
func Unwatch() (interface{}, error) {
	return mustGetClient().SendCommand("UNWATCH")
}
